assets/src/dashboard/app/api/useMediaApi.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo, useReducer } from 'react';
+import queryString from 'query-string';
+
+/**
+ * Internal dependencies
+ */
+import mediaReducer, {
+  defaultMediaState,
+  ACTION_TYPES as MEDIA_ACTION_TYPES,
+} from '../reducer/media';
+
+export default function useMediaApi(dataAdapter, { globalMediaApi }) {
+  const [state, dispatch] = useReducer(mediaReducer, defaultMediaState);
+
+  const fetchMediaById = useCallback(
+    async (mediaIds) => {
+      dispatch({
+        type: MEDIA_ACTION_TYPES.LOADING_MEDIA,
+      });
+
+      try {
+        const response = await dataAdapter.get(
+          queryString.stringifyUrl({
+            url: globalMediaApi,
+            query: {
+              include: mediaIds.join(','),
+              per_page: 100,
```

Comment:
We could theoretically do `per_page: -1` to load more than 100 items

Reply:
However, the database column for the publisher logos IDs in WordPress has some limit, so I don't think we'd ever reach such a high number.

Reply:
It seems like an odd case to have more than 100 publisher logos ever? We can loop through pages as we have set up for categories and tags. I'm also concerned about what this looks like on the page. Right now, how i have it set up if you have more than 100 it won't break, those just won't render.

Thoughts on leaving it as an unsolved edge case to put it on the back burner to get some UX about how to handle it so that it doesn't go unhandled in the long run?  I can make a bug ticket for it and tag as ux needed.

Reply:
Yeah I don't think we need to accommodate for such a case.